src/validators/index.ts

```diff
@@ -9,15 +9,23 @@ class InvalidPathError extends Error {
     }
 }
 
+class ValidationFailedError extends Error {
+    constructor(message: string) {
+        super(message)
+        this.name = 'ValidationFailedError'
+    }
+}
+
 const validators: Record<string, typeof validateSatelliteIdRoute> = {
     '/two-line-elements': validateSatelliteIdRoute,
     '/satellite/by-id': validateSatelliteIdRoute,
     '/satellite/by-name': validateSatelliteNameRoute
 }
 
-export default (req: Request, res: Response, next: NextFunction): void => {
+export default (req: Request, _: Response, next: NextFunction): ReturnType<NextFunction> => {
     const currentValidator = validators[req.path]
 
-    if (currentValidator) return currentValidator(req, res, next)
-    throw new InvalidPathError('Invalid Path')
+    if (!currentValidator) return next(new InvalidPathError('Invalid Path'))
+    if (!currentValidator(req)) return next(new ValidationFailedError('Validation Failed'))
+    return next()
```

Comment:
Which status code will a validation error result in? Since you haven't attached your own error handling middleware, I think validation errors will return HTTP 500. Consider adding a custom error handling middleware that checks if error = ValidationFailedError and responds with a more appropriate status code.